### ISSUE TITLE
PacBio seq full length ASV's Kingdome to strain level mapping 

### ISSUE THREAD

**Issue by sramani-spark** [benjjneb/dada2#1438]
Hi @benjjneb ,

I have digesta samples from 8 groups and 6 biological replicates but no technical replicates, which are sequenced in PacBio as 16s-full length, I carried the analysis with DADA2 pipeline [https://benjjneb.github.io/LRASManuscript/LRASms_fecal.html](url), 
I would like to know 
1) How shall carry the technical replicate part as I have the Biological replicates instead,
2) How shall I export the taxonomy which is assigned, the sequence and count table for every sample is not being able to export using the code : `write.table(tax2, "tax2.xlsx", sheetName = "Data_Frame")` for which I have installed the packages `"xlsx",` "xlsxjars" and `"openxlsx".`
3) Initially I had issues with plotting the Ecoli but after imposing `tax2[,"Genus"] <- gsub("Escherichia/Shigella", "Escherichia", tax2[,"Genus"])` which helped to reformat as compatible with other data sources so I was able to plot/see the data, but strain type is being assigned as Ec1, Ec2,..., rather, I would like to assign the actual strain nomenclature, for taxonomy assignment I have 17 species level assignment out of 414 entries rest 397 are showing as NA.
**Note: In my study I utilized 12 live probiotic strains as supplement for the animal treatment which are also not mapped**

Could you please help me in sorting the above said issues and please tell me if I need to provide any other information further.

![image](https://user-images.githubusercontent.com/15093709/138603309-08e823a8-3aeb-4621-ac02-562dd3cf6d1a.png)

**Comment by benjjneb** [comment]
1. You probably don't? The technical replicates in the analysis you linked was done for the purpose of validating PacBio CCS, and while can be useful is not needed every time.

2. I would use the `write.csv` function, and load the csv file into Excel, rather than trying to write a xlsx file directly. 

3. I don't really follow...

**Comment by SisuPark** [comment]
@benjjneb 
Thank you for the response and my apologies for not properly describing the issue. 

I'm new to the R programming and but still I followed your excellent protocols to for the analysis, let me try to explain again.

1) For exporting the taxonomy table or ASV count table "st2" I used `write.csv` which worked, but now I see the column names or row names as individual sequences itself rather than the taxonomy ranks.  I would like to use  conventional names rather for example ASV1, ASV2,.. could you please tell me how shall I do this both in sequence count table and taxonomy table.

2) Initially I used the [silva_nr99_v138.1_train_set.fa.gz](https://zenodo.org/record/4587955#.YX0wKByRWUk) for assigning the taxonomy so most of the taxonomy ranks were not properly assigned, but when I tried to use [silva_nr99_v138.1_wSpecies_train_set.fa.gz](https://zenodo.org/record/4587955#.YX0wKByRWUk) library half of the taxonomy ranks are assigned at species level, So I wonder if I can assign more when I use different database or strategy.
Is it possible to BLAST from R environment and dada2 itself for all other unassigned ASVs?

3)  After dada2 analysis I used your [Workflow for Microbiome Data Analysis: from raw reads to community analyses](https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html) for phylogeny assignment and tree construction. After this I wish to see the radiating nodes or final nodes with the taxonomy ranks or just as  ASV1, ASV2,.. can you please tell me how to do that? 

4) I want to export the tree coordinates like node and tips to visualize in other external tools for example [(iTOL)](https://itol.embl.de/),  I tried to use `ape::write.tree(fitGTR$tree, file='fitGTR$tree.txt') # text file with Newick format` or `ape::write.nexus(fitGTR, file='fitGTR.nex') # for Nexus format` which also has the sequence names as full sequence but not as taxonomy ranks or node ID. That will be really appreciative if you can tell how to export this with taxonomy ranks or convenient node IDs or names instead of entire sequence. 
Is it possible to have the circular visualization of phylogeny tree with colors? 

Thank you.

**Comment by benjjneb** [comment]
1. You can rename column names of the sequence table to be "ASV1"... etc instead of the actual sequences, and then export that modified table.

```
st.export <- st
colnames(st.export) <- paste0("ASV", seq(ncol(st)))
write.csv(st.export, ...)
```

2. When I looked for a nice R wrapper for BLAST a few years ago I didn't find anything I liked much. I just call command-line BLAST via `system` calls, or just work with BLAST on the command line directly. Maybe there are better BLAST R packages out there now?

3. This is outside the DADA2 R package, and you'll want to look for assistance with the appropriate software package, or the Github repo associated with the manuscript.

4. As above.
